### PR TITLE
Add git commands 'stageAll', 'unstageAll', and 'discardAll'

### PR DESCRIPTION
**Pull request by vince-fugnitto**
- Added the git commands `stageAll`, `unstageAll` and `discardAll` to the `git-widget`.

Signed-off-by: Vincent Fugnitto <vincent.fugnitto@ericsson.com>

### CODE REVIEW

**Comment by jbicker** [comment]
'Stage all' and 'Unstage all' works fine so far.
'Discard all' does not work for new files. They don't get removed as expected.
Furthermore I would like it to have the buttons in the 'Changes' resp. 'Staged changes' header lines with the same hover behaviour as for single files. See VSCode.
And for 'Discard' you should use the same Icon as it is used for discarding single changes.

**Comment by akosyakov** [comment]
In additional to @jbicker's [comment](https://github.com/theia-ide/theia/pull/3943#issuecomment-451387924) it would be nice to add these action in the drop-down menu.

**Comment by vince-fugnitto** [comment]
@jbicker @akosyakov 
thanks guys, I'll follow what `vscode` has as a ui/ux experience for these commands and update the PR accordingly.

**Comment by vince-fugnitto** [comment]
@jbicker @akosyakov 
I made the necessary updates both adding the inline toolbar to the `Staged Changes`, `Merged Changes` and `Changes`, as well as adding the new menu items in the context menu for the `git-widget`. I also addressed the issue regarding the `discardAll` for newly created files. Please let me know if anything else is needed. @jbicker perhaps you can help in making the toolbar for the header items look better.

File: packages/git/src/browser/git-widget.tsx [hunk 1]
```diff
@@ -821,9 +900,10 @@ export class GitChangesListContainer extends React.Component<GitChangesListConta
     protected renderUnstagedChanges(repository: Repository | undefined): React.ReactNode | undefined {
         if (this.props.unstagedChanges.length > 0) {
             return <div id='unstagedChanges' className='changesContainer'>
-                <div className='theia-header'>
+                <div className='theia-header git-theia-header'>
                     Changes
                     {this.renderChangeCount(this.props.unstagedChanges.length)}
+                    {this.renderChangeListButtons()}
```
Comment (jbicker) [inline packages/git/src/browser/git-widget.tsx]: You should only render the 'stage all' button and in addition the 'discard all' button here.

File: packages/git/src/browser/git-widget.tsx [hunk 2]
```diff
@@ -793,9 +870,10 @@ export class GitChangesListContainer extends React.Component<GitChangesListConta
     protected renderMergeChanges(repository: Repository | undefined): React.ReactNode | undefined {
         if (this.props.mergeChanges.length > 0) {
             return <div id='mergeChanges' className='changesContainer'>
-                <div className='theia-header'>
-                    Merge Changes
+                <div className='theia-header git-theia-header'>
+                    Merged Changes
```
Comment (jbicker) [inline packages/git/src/browser/git-widget.tsx]: You should only render the 'stage all' button here.

**Comment by vince-fugnitto** [comment]
thanks @jbicker for looking into it and providing feedback!

**Pull request merged**